media-source-respec.html

```diff
@@ -2720,6 +2712,280 @@ <h3>Event Summary</h3>
+        <thead>
+          <tr>
+            <th>Event name</th>
+            <th>Interface</th>
+            <th>Dispatched when...</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              <dfn class="event">bufferedchange</dfn>
+            </td>
+            <td>{{Event}}</td>
+            <td>
+              The {{ManagedSourceBuffer}}'s buffered range changed following a
+              call to {{SourceBuffer/appendBuffer()}}, {{SourceBuffer/remove()}},
+              or as a consequence of the user agent running the
+              [=ManagedSourceBuffer/memory cleanup=] algorithm.
+            </td>
+          </tr>
+        </tbody>
+      </table>
+
+      <h3>Algorithms</h3>
+      <h4>Buffered Change</h4>
+      <p>
+        The following steps are run at the completion of all operations to the {{ManagedSourceBuffer}}
+        |buffer:ManagedSourceBuffer| that would cause a |buffer|'s {{SourceBuffer/buffered}} to change.
+        That is once {{SourceBuffer/appendBuffer()}}, {{SourceBuffer/remove()}} or
+        [=ManagedSourceBuffer/memory cleanup=] algorithm have completed.
+      </p>
+      <ol>
+        <li>
+          Let |previous buffered ranges:normalized TimeRanges| equal the
+          {{SourceBuffer/buffered}} attribute before the changes occurred.
+        </li>
+        <li>
+          Let |new buffered ranges:normalized TimeRanges| equal the new
+          {{SourceBuffer/buffered}} {{TimeRanges}}.
+        </li>
+        <li>
+          Let |added:normalized TimeRanges| equal the |previous buffered
+          ranges:normalized TimeRanges| subtracted from |new buffered
+          ranges:normalized TimeRanges|.
+        </li>
+        <li>
+          Let |removed:normalized TimeRanges| equal the |new buffered
+          ranges:normalized TimeRanges| subtracted from |previous buffered
+          ranges:normalized TimeRanges|.
+        </li>
+        <li>
+          Let |eventInitDict| be a new {{BufferedChangeEventInit}} dictionary
+          initialized with |added| as its {{BufferedChangeEventInit/addedRanges}} and
+          |removed| as its {{BufferedChangeEventInit/removedRanges}}
+        </li>
+        <li>
+          [=Queue a task=] to [=fire an event=] named {{bufferedchange}} at |buffer|
+          using the {{BufferedChangeEvent}} interface, initialized with
+          |eventInitDict|.
+        </li>
+      </ol>
+      <h4>
+        <dfn data-dfn-for="ManagedSourceBuffer">Memory cleanup</dfn>
+      </h4>
+      <ol>
+        <li>
+          <dl class="switch">
+            <dt>
+              If [=this=] is not in [=this=]'s {{ManagedMediaSource}} parent
+              {{MediaSource/activeSourceBuffers}}:
+            </dt>
+            <dd>
+              <ol>
+                <li>
+                  Run the [=coded frame removal=] algorithm with 0 and positive infinity equal to the removal range start and end timestamp respectively, and abort these steps.
```

Comment:
A minor not, for consistency with how this is phrased elsewhere in the spec I'd suggest phrasing as "run the coded frame removal algorithm with start and end equal to 0 and positive infinity respectively.